yapapi/rest/activity.py

```diff
@@ -197,6 +199,16 @@ async def __aiter__(self) -> AsyncIterator[events.CommandEventContext]:
             except ApiException as err:
                 if err.status == 408:
                     continue
+                if err.status == 500:
+                    try:
+                        body = json.loads(err.body)
+                        if 'GSB error' in body['message']:
+                            if retry_count < MAX_RETRIES:
+                                retry_count += 1
+                                await asyncio.sleep(min(3, max(0, self.seconds_left())))
+                                continue
+                    except:
+                        pass
```

Comment:
A couple of remarks:
* This adds a lot of complication to this function (`try-catch` statement nested inside `except` clause etc.)
* Shouldn't we reset `retry_count` each time a successful call to `get_exec_batch_results()` is made?
* Why retry only 3 times on GSB errors while we do retry until batch timeout on timeout errors?

I would just decorate the definition of `get_exec_batch_results()` using the `repeat_on_error` decorator with a `condition` that detects timeout errors and this new GSB error. That would address all three remarks.